docs/source/tools/trigger-service.rst

```diff
@@ -0,0 +1,119 @@
+
+The above starts the trigger-service using a number of default parameters. Most notably, the HTTP port the trigger service listens on which defaults to 8088. The meanining of the ``--no-secret-key`` parameter will be explained in the next section. To see all of the available parameters, their defaults and descriptions, one can execute the command ``daml trigger-service --help``.
+
+Although as we'll see, the trigger-service exposes an endpoint for end-users to upload DAR files to the service (DAML archives containing triggers), it is sometimes convenient to start the service preconfigured with a specific DAR. To do this, the ``--dar`` option is provided.
+
+.. code-block:: bash
+
+   daml trigger-service --ledger-host localhost --ledger-port 6865 --wall-clock-time --no-secret-key --dar <DAR>
+
+Identifying trigger parties
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this section, we briefly explain how parties are identified in end-user interactions with a running trigger service.
+
+When an end-user interacts with a trigger-service, they do so by sending HTTP requests to one of several trigger-service HTTP endpoints. Such requests must contain information detailing the party the trigger-service is to act on the behalf of in its handling of the user request.
+
+This party information is conveyed via  HTTP basic authentication headers where the requesting party is represented by a unique username/password pair (Later versions of this documentation will explain running the trigger-service in authenticated contexts. For now, we assume non-authenticated contexts. In these contexts, the passwords associated with parties are not important and may have any values at all but they still must be provided).
+
+With an understanding of the role of credentials now in mind we can explain the ``--no-secret-key`` parameter passed to the trigger-service in the preceding section. When the trigger-service stores credentials it first encrypts them. The encryption/decryption algorithms it uses to do so require the use of a "key". What key is to be used is expected to be provided as the value of the environment variable ``TRIGGER_SERVICE_SECRET_KEY`` (for example, before starting the trigger-service by a command like ``export TRIGGER_SERVICE_SECRET_KEY="my secret key"``). At startup, if the trigger-service detects that the environment variable is not defined then it will terminate unless the parameter ``--no-secret-key`` is in its startup options. If it is, the service will continue with a default key.
+
+If the party 'alice' has password '&alC2l3SDS*V' then we can construct her credentials token by joining her username to her password separated by ':' giving 'alice:&alC2l3SDS*V'. Then base64 encoding her credentials token gives 'YWxpY2U6JmFsQzJsM1NEUypW'. We will see example usage of the credential tokens produced in this fashion in the next section where examples are given sending requests to the trigger-service.
```

Comment:
I don’t think this is the right place to explain how HTTP Basic Auth works. Basically all http libraries do this for you. Even `curl` does! You can simplify your examples below by using curl’s `--user` option and nobody ever has to know that this is base64 encoded or which header you have to set.

Reply:
i didn't know about that. ok, let me look into this and removing this crud :)

Reply:
✔️